api/envoy/config/common/matcher/v3/matcher.proto

```diff
@@ -140,6 +140,17 @@ message Matcher {
   OnMatch on_no_match = 3;
 }
 
+// Wrapper around an existing extension that provides an associated matcher. This allows
+// decorating an existing extension with a matcher, which can be used to match against
+// relevant protocol data.
+message ExtensionWithMatcher {
```

Comment:
Is this the right location for this message?  It seems like this message is more about extensions than about matchers, so maybe it would make more sense to put it somewhere with other extension-related messages?

Reply:
Good point. I don't think I can put this in the same package as TypedExtensionConfig since there would be a circular reference between this and that package, but maybe I can make a new one in envoy::config::core to hold this? I'll see what it looks like

Reply:
+1 I agree with Mark. A new location SGTM.